srcdeps-core/pom.xml

```diff
@@ -74,17 +74,32 @@
       <scope>test</scope>
     </dependency>
 
-    <dependency>
-      <groupId>org.srcdeps.core</groupId>
-      <artifactId>srcdeps-core-gradle-settings</artifactId>
-      <type>gradle</type>
-    </dependency>
-
   </dependencies>
 
   <build>
     <plugins>
 
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-antrun-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>antrun-copy</id>
+            <phase>process-resources</phase>
+            <configuration>
+              <target name="move SrcdepsInner.java to SrcdepsInner.gradle">
```

Comment:
Wait, what? O_O
What is it doing? :)

Reply:
Yeah, this needs some explanation. I tried several ways to set versions in an inner gradle build: (a) gradle script parsing and subsequent transformation did not work, (b) injecting a byteman script into the inner build process worked, but then, @uklance suggested (thanks!) on gradle forums [1] that it should be possible to inject a piece of script to `settings.gradle` to do that. That piece of script lives in `srcdeps-transform.gradle` [2]. It is fairly simple but that piece of script needs some data from the outer build. `SrcdepsInner` in the main source tree encompasses those and needs to be injected to `settings.gradle` too. And now, back to your question: on one hand, I hate to maintain groovy/gradle scripts and on the other hand the greater part of that code exists a Java code already anyway. So what I do is just to take the needed java files, copy them as `.groovy` to resources so that I can pull them from there when injecting them to `settings.gradle`. Does it make sense?

[1] https://discuss.gradle.org/t/a-programatic-way-to-set-the-version-of-any-gradle-project/23197/5

[2] https://github.com/srcdeps/srcdeps-core/pull/15/commits/c5c5a38c00cf1e58c70c77919de15b79aad4bb85#diff-629c7cdee9c8a49d11bbd80c33819079

Reply:
Wow, that's sorcery! I like it, but leave a comment on the code explaining it, perhaps linking to your comment above.

Reply:
Yes, a comment is undeway in the next PR.